UPGRADE-4.0.md

```diff
@@ -0,0 +1,98 @@
+$response =  $matej->request()
+    ->sorting(Sorting::create('user-id', ['item-id-1', 'item-id-2', 'item-id-3']))
+    ->send();
+```
+
+#### After
+```php
+$response =  $matej->request()
+    ->sorting(ItemSorting::create('user-id', ['item-id-1', 'item-id-2', 'item-id-3']))
+    ->send();
+```
+
+
+## New types of recommendations
+API client now supports new types of recommendation commands:
+- Recommend items to user (`UserItemRecommendation`, originally `UserRecommendation`)
+- Recommend users to user (`UserUserRecommendation`)
+- Recommend items to item (`ItemItemRecommendation`)
+- Recommend users to item (`ItemUserRecommendation`)
+
+Each recommendation type supports different parameter options. Awailable parameters are
+described in table bellow:
+
+| Atribute      | Methods                                   | UserItemRecommendation | UserUserRecommendation | ItemItemRecommendation | ItemUserRecommendation |
+|---------------|-------------------------------------------|------------------------|------------------------|------------------------|------------------------|
+| scenario      |              `static::create`               |            ✅           |            ✅           |            ✅           |            ✅           |
+| count         |                  `setCount`                 |            ✅           |            ✅           |            ✅           |            ✅           |
+| rotation_rate |              `setRotationRate`              |            ✅           |            ✅           |            ❌           |            ❌           |
+| rotation_time |              `setRotationTime`              |            ✅           |            ✅           |            ❌           |            ❌           |
+| hard_rotation |             `enableHardRotation`            |            ✅           |            ✅           |            ❌           |            ❌           |
+| allow_seen    |                `setAllowSeen`               |            ✅           |            ❌           |            ❌           |            ✅           |
+| min_relevance |            `setMinimalRelevance`            |  `ItemMinimalRelevance`  |            ❌           |            ❌           |  `UserMinimalRelevance`*  |
+| filter        |            `addFilter` `setFilters`           |            ✅           |            ❌           |            ✅           |            ❌           |
+| boost_rules   |             `addBoost` `setBoosts`            |            ✅           |            ❌           |            ✅           |            ❌           |
+| model_name    |                `setModelName`               |            ✅           |            ✅           |            ✅           |            ✅           |
+| properties    | `addResponseProperty` `setResponseProperties` |            ✅           |            ❌           |            ✅           |            ❌           |
+
+\* `UserMinimalRelevance` supports only `MEDIUM` and `HIGH` relevancies.
```

Comment:
```suggestion
\* `UserMinimalRelevance` supports only `MEDIUM` and `HIGH` relevances.
```

Also this part is missing in the readme (so I suggest moving it to readme only).